Implement 1D PDFs in B->Vlnu code

dvandyk opened the pull request:
- [x] Implement the 1D PDFs in B->Vlnu code and copy existing test cases
- [x] Remove old B->Dpilnu code
- [x] Understand why the new code violates old test cases by up to 3.0e-2

@MartinJung1978 Can you and I compare predictions for the B->D*lnu PDFs in the week starting April 28th? The old, violated test cases claim agreement with your notebook.

dvandyk committed:
[b-decays] Convert B->D^*lnu 1D PDFs to use ``BToVectorLeptonNeutrino``

dvandyk committed:
[b-decays] Remove old B->Dpilnu code

mreboud commented on eos/b-decays/b-to-vec-l-nu.cc:
```diff
@@ -341,6 +341,72 @@ namespace eos
+
+            return 0.5 * (c_theta_l_max - c_theta_l_min)
+                + a_fb * 0.5 * (c_theta_l_max * c_theta_l_max - c_theta_l_min * c_theta_l_min)
+                + (1.0 - 3.0 * ftilde_L) / 8.0 * ((c_theta_l_max * c_theta_l_max * c_theta_l_max - c_theta_l_min * c_theta_l_min * c_theta_l_min) - (c_theta_l_max - c_theta_l_min));
```
Can you use `power_of<n>` for n > 2? I find it easier to read.

mreboud commented on eos/b-decays/b-to-vec-l-nu.cc:
```diff
@@ -341,6 +341,72 @@ namespace eos
+
+            return 3.0 / 4.0 * (
+                (1.0 - f_L) * (c_theta_v_max - c_theta_v_min)
+                - (1.0 - 3.0 * f_L) * (c_theta_v_max * c_theta_v_max * c_theta_v_max - c_theta_v_min * c_theta_v_min * c_theta_v_min) / 3.0
```
Here as well

dvandyk replied:
Fixed.

mreboud reviewed (commented):
All good so far!

dvandyk committed:
[b-decays] Regenerate B->D^*lnu 1D-PDF reference values and tighten tolerance

The 1D-PDF reference values in the "check 1D PDFs" block were carried over
verbatim from the removed BToDPiLeptonNeutrino test and only held at a loose
tolerance (3e-2). Comparison against [DDS:2014A] (arXiv:1405.3719) shows:

  - the old azimuthal (chi) distribution was incorrect: it contained a
    spurious linear cos(chi) term (proportional to sin(2 theta_D)), which
    vanishes upon integration over theta_D. Only the cos(2 phi) [J_3] and
    sin(2 phi) [J_9] terms survive, as implemented by
    BToVectorLeptonNeutrino::differential_pdf_phi.
  - the old differential PDFs used a hard-coded upper limit q2_max = 10.68
    instead of the correct (m_B - m_D*)^2 = 10.689, biasing the normalization.

Regenerate all values from BToVectorLeptonNeutrino and restore a tight 1e-5
tolerance. For this form-factor point J_3 ~= 0 and J_9 = 0, so the phi
distribution is flat (1 / (2 pi)).

dvandyk commented on eos/b-decays/observables.cc:
```diff
@@ -1628,54 +1627,60 @@ namespace eos
                                 Options{ { "V"_ok, "D^*"_ov } }),
 
                 // B -> D pi l nu
```
Fixed.

dvandyk commented on eos/b-decays/observables.cc:
```diff
@@ -1628,54 +1627,60 @@ namespace eos
-                                std::make_tuple("cos(theta_D)")),
+                                &BToVectorLeptonNeutrino::differential_pdf_w,
+                                std::make_tuple("w"),
+                                Options{ { "U"_ok, "c"_ov }, { "I"_ok, "1/2"_ov } }),
```
Fixed.

dvandyk committed:
[b-decays] Make B->D^*lnu angular 1D PDFs bin-averaged for consistency

The angular 1D-PDF observables P(cos(theta_D)_min,..), P(cos(theta_l)_min,..)
and P(phi_min,..) returned bin-integrated probabilities, whereas every other
PDF-like observable in the code base (all q^2/w projections across b-, c-, s-
and tau-decays, and the SignalPDF framework) uses a bin-averaged density.

Divide integrated_pdf_{l,v,phi} by their bin width so they return a density,
consistent with the corresponding differential_pdf_{l,v,phi}. integrated_pdf_q2
and integrated_pdf_w were already bin-averaged and are unchanged.

Rescale the Belle:2017A angular kinematical distributions accordingly: each
angular data point (mean and statistical/systematic uncertainties) is divided
by its bin width (Delta cos(theta) = 0.2, Delta phi = 2 pi / 10). The
correlations are invariant under this diagonal rescaling and are left
unchanged; the w bins and all other constraints are unaffected.

dvandyk commented on eos/b-decays/b-to-vec-l-nu.cc:
```diff
@@ -339,6 +339,72 @@ namespace eos
+
+            return 0.5 * (c_theta_l_max - c_theta_l_min)
+                + a_fb * 0.5 * (c_theta_l_max * c_theta_l_max - c_theta_l_min * c_theta_l_min)
+                + (1.0 - 3.0 * ftilde_L) / 8.0 * ((power_of<3>(c_theta_l_max) - power_of<3>(c_theta_l_min)) - (c_theta_l_max - c_theta_l_min));
```
This is a good point. I have adapted all of the PDFs to be bin-averaged rather than bin-integrated. This change also touched all constraints that reference these observables.

dvandyk committed:
[eos] Update changelog for PR #1006

Herren reviewed (approved):
I do not see any problems here. Happy to have the `DPi` gone :)
